test/python/topology/test_functions.py

```diff
@@ -52,56 +52,112 @@ def string_to_int_except68(t):
+   if _hwcount_transform == 0:
       if t != 342:
          raise AssertionError()
-      _hwcount += 1
+      _hwcount_transform += 1
       return None
-   if _hwcount == 1:
+   if _hwcount_transform == 1:
       if t != 474:
          raise AssertionError()
-      _hwcount += 1
+      _hwcount_transform += 1
       return None
-   if _hwcount == 2:
+   if _hwcount_transform == 2:
       if t != 9342:
          raise AssertionError()
-      _hwcount += 1
+      _hwcount_transform += 1
       return None
    raise AssertionError() 
 
+global _hwcount_transform_with_drop
+_hwcount_transform_with_drop = 0
 def check_int_strings_transform_with_drop(t):
    print("TUPLE", t)
-   global _hwcount
-   if _hwcount == 0:
+   global _hwcount_transform_with_drop
+   if _hwcount_transform_with_drop == 0:
       if t != 110:
          raise AssertionError()
-      _hwcount += 1
+      _hwcount_transform_with_drop += 1
       return None
-   if _hwcount == 1:
+   if _hwcount_transform_with_drop == 1:
       if t != 238:
          raise AssertionError()
-      _hwcount += 1
+      _hwcount_transform_with_drop += 1
       return None
    raise AssertionError() 
  
 def strings_multi_transform():
    return ["mary had a little lamb", "its fleece was white as snow"]
    
-global expected_strings_multi_transform
-expected_strings_multi_transform = ["mary", "had", "a", "little", "lamb", "its", "fleece", "was", "white", "as", "snow"]
+global EXPECTED_STRINGS_MULTI_TRANSFORM
+EXPECTED_STRINGS_MULTI_TRANSFORM = ["mary", "had", "a", "little", "lamb", "its", "fleece", "was", "white", "as", "snow"]
 
 def split_words(t):
    return t.split()
  
+global _hwcount_multi_transform
+_hwcount_multi_transform = 0
 def check_strings_multi_transform(t):
    print("TUPLE", t)
-   global _hwcount
-   assert (_hwcount < len(expected_strings_multi_transform)), \
-      ("Expected index=" + str(_hwcount) + " < " + str(len(expected_strings_multi_transform)))
-   assert (t == expected_strings_multi_transform[_hwcount]), \
-      ("Expected=" + expected_strings_multi_transform[_hwcount] +  ", actual=" + str(t))
-   _hwcount += 1
+   global _hwcount_multi_transform
+   assert (_hwcount_multi_transform < len(EXPECTED_STRINGS_MULTI_TRANSFORM)), \
+      ("Expected index=" + str(_hwcount_multi_transform) + " < " + str(len(EXPECTED_STRINGS_MULTI_TRANSFORM)))
+   assert (t == EXPECTED_STRINGS_MULTI_TRANSFORM[_hwcount_multi_transform]), \
+      ("Expected=" + EXPECTED_STRINGS_MULTI_TRANSFORM[_hwcount_multi_transform] +  ", actual=" + str(t))
+   _hwcount_multi_transform += 1
+   
+def strings_length_filter():
+   return ["hello", "goodbye", "farewell"]
+
+class LengthFilter:
+   def __init__(self, upper):
+      self.upper = upper  
+   def __call__(self, tuple):
+      if len(tuple) > self.upper:
+         return True
+      return False
+
+global EXPECTED_STRINGS_LENGTH_FILTER
+EXPECTED_STRINGS_LENGTH_FILTER = ["goodbye", "farewell"]
+
+global _hwcount_length_filter
```

Comment:
@wmarshall484 
Currently I define a global counter for each new testcase.  To avoid this, do you think it makes sense to support parameters for sink() as well (maybe source() too) ?  Thanks.